datafusion/core/src/execution/session_state.rs

```diff
@@ -293,7 +293,9 @@ impl SessionState {
             .resolve(&catalog.default_catalog, &catalog.default_schema)
     }
 
-    pub(crate) fn schema_for_ref(
+    /// Retrieve the [`SchemaProvider`] for a specific [`TableReference`], if it
+    /// esists.
+    pub fn schema_for_ref(
```

Comment:
Not sure if there is a way in which we could keep this method as `pub(crate)`, but I don't see many reasons to make this method not public.